modules/data.land/R/soil_params_ensemble.R

```diff
@@ -0,0 +1,239 @@
+##' A function to estimate individual alphas for Dirichlet distribution to approximate the observed quantiles with means as known moments
+##' for SoilGrids soil texture data. 
+##' Dirichlet distribution is assumed as soil texture data follow categorical distribution and the probability of each category is in the range 0 to 1, 
+##' and all must sum to 1.
+##'  
+##' @param means A vector of means of sand, clay, and silt proportion for one soil layer at one site from SoilGrids data
+##' @param quantiles A list of 5th, 50th, and 95th percentiles for sand, clay, and silt for one soil layer at one site from SoilGrids data
+##'
+##' @example
+##' \dontrun{
+##' means <- c(0.566,0.193,0.241) # means of sand, clay, and silt at one site and one depth
+##' quantiles <-list(
+##' q5 = c(0.127,0.034,0.052), # 5th percentile for each category: sand, clay, and silt at one site and one depth
+##' q50 = c(0.615,0.15,0.191), # 50th percentile (median) for each category: sand, clay, and silt at one site and one depth
+##' q95 = c(0.799,0.66,0.616))  # 95th percentile for each category: sand, clay, and silt at one site and one depth
+##' alpha_est <- estimate_dirichlet_parameters(means, quantiles)
+##' }
+##'  @return The individual alphas that work best to fit the observed quantiles
+##'  @export 
+##'  @author Qianyu Li
+estimate_dirichlet_parameters <- function(means, quantiles) {
```

Comment:
I don't think it should be combined with the existing function `prior_fn` (that is similar in approach but for univariate distributions though it has its own limitations). But perhaps it should be included in the same file in the priors package?

https://github.com/PecanProject/pecan/blob/edc14b81a8da4fe71f93602137ddf0dd0308a4d1/modules/priors/R/priors.R#L90